### ISSUE TITLE
Javascript error in IE11

### ISSUE THREAD

**Issue by xavadu** [MTobisch/ngx-dynamic-hooks#2]
Hello,

I am having the following error in IE11 with ES5 `ERROR SyntaxError: Syntax error in regular expression` with the following polyfill

```
import 'web-animations-js';
import 'web-animations-js';
import 'zone.js/dist/zone';

import 'core-js/es/object';
import 'core-js/es/array';
import 'core-js/es/string';
import 'core-js/es/regexp';
```

Any idea if I am missing some polyfill to make it work? or could be an code issue? Is quite complicate to debug in IE

Thanks

**Comment by MTobisch** [comment]
Hi there,

to be honest, I haven’t really tested the library on older browsers, so that’s the first time I’ve seen that error. I’m also having trouble reproducing the result as I can’t even get IE to run with normal Angular or Stackblitz properly. I quick Google search returns me mostly these two results:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Bad_regexp_flag

https://github.com/vercel/next.js/discussions/10870

I’m not sure either is helpful to you. Here’s what I can tell you about the code of the library, though: For all my regex I use standard RegExp objects<https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp> instantiated by their constructor (which should be compatible with IE11) and I’m also not using the ‘y’-flag that seems to be causing problems sometimes. It’s all fairly harmless stuff.

Are you perhaps using custom hook parsers with your own regex that could be causing this? Otherwise, I’m afraid I’m at a loss as well.

All the best,
Marvin

**Comment by xavadu** [comment]
I was googling some hour before report a second issue here ;p and I was in some of this resources

I think that the issue could come from here https://stackoverflow.com/a/40629349 IE11 do not support the syntax of regular expression literal like `/a|regular|expr/` it need to be defined as a string `"a|regular|expr"`

So far I did not found a  regular expression literal in code but maybe it give you a hit, the error message is the same in IE11:

![image](https://user-images.githubusercontent.com/1382820/103347330-f7549f00-4a96-11eb-9e74-b37292a1aabf.png)

**Comment by axmad22** [comment]
@MTobisch  You just gave flat Angular some Breasts nice work Doc,...am looking forward in testing them :) thanks

**Comment by MTobisch** [comment]
@xavadu I've taken another look at the problem and released a new version of this library to address it: 1.4.1.

The problem wasn't so much that I used regex literals in the RegExp constructor. It was that I used the "s" flag in some of my regexes, which I didn't realize is a pretty new thing.

However, fixing that only gave me another (logically unconnected) error due to it being IE11. So I went down the rabbit hole a bit and fixed a bunch of stuff until it ran without issue for me. I hope this new version gets rid of all the errors you might have with IE11 as well.
